Yatzy-Kata/Yatzy.Console/Strategies/ConsoleUserStrategy.cs

```diff
@@ -0,0 +1,66 @@
+using Yatzy_Kata.Data;
+using Yatzy_Kata.Interfaces;
+
+namespace Yatzy_Kata.Strategies;
+
+public class ConsoleUserStrategy : IStrategy
+{
+    private readonly ConsoleDiceRollStrategy _diceRollStrategy;
+    private readonly IReader _reader;
+    private readonly IWriter _writer;
+    private bool _abandoned;
+    private TurnResults? _turnResults;
+
+    public ConsoleUserStrategy(IReader reader, IWriter writer, IRandom random)
+    {
+        _reader = reader;
+        _writer = writer;
+        _abandoned = false;
+        _diceRollStrategy = new ConsoleDiceRollStrategy(reader, writer, random);
+    }
+    
+    public TurnResults? GetTurnResults()
+    {
+        return _turnResults;
+    }
+
+    public void CalculateTurnResults(List<Category> remainingCategories)
```

Comment:
I notice that this method is the same as in this class `Yatzy-Kata/Yatzy.Core/ComputerStrategies/ComputerStrategy.cs`. Can you think of a way to avoid this?

Reply:
Hi Michelle! Thank you for the comment. I will update my readMe regarding my choice of pattern and about better ways i could of designed this kata. About the code duplication thank you for spotting this out, initially those two methods were different but as i abstracted them both I completely missed that they became the same thing. will update!